pkg/apis/policy/validation/validation.go

```diff
@@ -78,6 +80,17 @@ func ValidatePodDisruptionBudgetSpec(spec policy.PodDisruptionBudgetSpec, opts P
 
 	allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(spec.Selector, labelSelectorValidationOptions, fldPath.Child("selector"))...)
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.PDBPodHealthyPolicy) {
```

Comment:
You want to validate the values always, otherwise you'll end up with a resource with invalid values which then won't be able to run when the feature is on.